mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_service_account_jwt.go

```diff
@@ -0,0 +1,145 @@
+			"expires_in": schema.Int64Attribute{
+				Optional:    true,
+				Description: "Number of seconds until the JWT expires. If set and non-zero an `exp` claim will be added to the payload derived from the current timestamp plus expires_in seconds.",
+			},
+			"target_service_account": schema.StringAttribute{
+				Description: "The email of the service account that will sign the JWT.",
+				Required:    true,
+			},
+			"delegates": schema.SetAttribute{
+				Description: "Delegate chain of approvals needed to perform full impersonation. Specify the fully qualified service account name.",
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+			"jwt": schema.StringAttribute{
+				Description: "The signed JWT containing the JWT Claims Set from the `payload`.",
+				Computed:    true,
+				Sensitive:   true,
+			},
+		},
+	}
+}
+
+func (p *googleEphemeralServiceAccountJwt) Configure(ctx context.Context, req ephemeral.ConfigureRequest, resp *ephemeral.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	pd, ok := req.ProviderData.(*fwtransport.FrameworkProviderConfig)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *fwtransport.FrameworkProviderConfig, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	p.providerConfig = pd
+}
+
+func (p *googleEphemeralServiceAccountJwt) Open(ctx context.Context, req ephemeral.OpenRequest, resp *ephemeral.OpenResponse) {
+	var data ephemeralServiceAccountJwtModel
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	payload := data.Payload.ValueString()
+
+	if !data.ExpiresIn.IsNull() {
+		expiresIn := data.ExpiresIn.ValueInt64()
+		if expiresIn != 0 {
+			var decoded map[string]interface{}
+			if err := json.Unmarshal([]byte(payload), &decoded); err != nil {
+				resp.Diagnostics.AddError("Error decoding payload", err.Error())
+				return
+			}
+
+			decoded["exp"] = time.Now().Add(time.Duration(expiresIn) * time.Second).Unix()
+
+			payloadBytesWithExp, err := json.Marshal(decoded)
+			if err != nil {
+				resp.Diagnostics.AddError("Error re-encoding payload", err.Error())
+				return
+			}
+
+			payload = string(payloadBytesWithExp)
+		}
+	}
+
+	name := fmt.Sprintf("projects/-/serviceAccounts/%s", data.TargetServiceAccount.ValueString())
+
+	service := p.providerConfig.NewIamCredentialsClient(p.providerConfig.UserAgent)
+	DelegatesSetValue, _ := data.Delegates.ToSetValue(ctx)
+	jwtRequest := &iamcredentials.SignJwtRequest{
+		Payload:   payload,
+		Delegates: StringSet(DelegatesSetValue),
+	}
+
+	jwtResponse, err := service.Projects.ServiceAccounts.SignJwt(name, jwtRequest).Do()
+	if err != nil {
+		resp.Diagnostics.AddError("Error calling iamcredentials.SignJwt", err.Error())
+		return
+	}
+
+	data.Jwt = types.StringValue(jwtResponse.SignedJwt)
+
+	resp.Diagnostics.Append(resp.Result.Set(ctx, data)...)
+}
+
+func StringSet(d basetypes.SetValue) []string {
```

Comment:
Is this the function that we moved to another package? Can we remove it and use that package?